src/main/kotlin/com/criptext/mail/scenes/emaildetail/workers/ReadEmailsWorker.kt

```diff
@@ -27,25 +28,63 @@ class ReadEmailsWorker(private val dao: EmailDao,
 
     override val canBeParallelized = false
     private val apiClient = EmailDetailAPIClient(httpClient, activeAccount.jwt)
+    private var postEmailsRead = 0
+    private var peerEmailsRead = 0
```

Comment:
don't keep state here, you could easily return these values from `postOpenEmails()` and `peerOpenEmails()`. Also you should test this.